core/src/main/java/io/seata/core/store/db/sql/lock/AbstractLockStoreSql.java

```diff
@@ -149,13 +170,75 @@ public String getQueryLockSql(String lockTable) {
     }
 
     @Override
-    public String getCheckLockableSql(String lockTable, String paramPlaceHold) {
-        return CHECK_LOCK_SQL.replace(LOCK_TABLE_PLACE_HOLD, lockTable).replace(IN_PARAMS_PLACE_HOLD, paramPlaceHold);
+    public String getCheckLockableSql(String lockTable, int rowSize) {
+        List<String> pkNameList = new ArrayList<>();
+        pkNameList.add(ServerTableColumnsName.LOCK_TABLE_ROW_KEY);
+        String whereCondition = buildWhereConditionByPKs(pkNameList,rowSize,MAX_IN_SIZE);
+        return CHECK_LOCK_SQL.replace(LOCK_TABLE_PLACE_HOLD, lockTable).replace(LOCK_TABLE_PK_WHERE_CONDITION_PLACE_HOLD, whereCondition);
     }
 
     @Override
     public String getBatchUpdateStatusLockByGlobalSql(String lockTable) {
         return BATCH_UPDATE_STATUS_LOCK_BY_GLOBAL_SQL.replace(LOCK_TABLE_PLACE_HOLD, lockTable);
     }
 
+    /**
+     * each pk is a condition.the result will like :" (id,userCode) in ((?,?),(?,?)) or (id,userCode) in ((?,?),(?,?)
+     * ) or (id,userCode) in ((?,?))"
+     * Build where condition by pks string.
+     *
+     * @param pkNameList pk column name list
+     * @param rowSize    the row size of records
+     * @param maxInSize  the max in size
+     * @return return where condition sql string.the sql can search all related records not just one.
+     */
+    private String buildWhereConditionByPKs(List<String> pkNameList, int rowSize, int maxInSize) {
```

Comment:
similar with `SqlGenerateUtils`, whether can extract a public util class?

Reply:
I tried to extract `SqlGenerateUtils` from rm module to the common module, it can be referenced by rm and core ,but  it also depended on  `ColumnUtils` class, I'm not sure how to deal with it